### DISCUSSION TITLE
Importing data from another lazer dir

### DISCUSSION THREAD

**Discussion by orangixx** [ppy/osu#13346] [Ideas] [answered]
Using the built in file browser, users should be able to select another osu!lazer directory and import data without overwriting the entire current directory.

(May or may not have deleted mine by accident, then realized there was a backup.)

**Comment by peppy** [comment, marked as answer]
You can just replace the file system. Merging lazer installs is out of scope.

**Reply by orangixx** [reply]
If I want to merge 2 installs' data, I have to dump install 1's client.db, remove duplicates, change every item ID, read from my modified dumpfile into install 2's client.db, and combine the 2 files/ directories.

It is already possible to import stable scores and beatmaps (even from exotic locations), why should there not be support for importing from another lazer install? This is definitely a backlog feature, but I believe it is still one that should be implemented.

BeatmapInfo, ScoreInfo, ScoreFileInfo, and a few other tables appear to contain hashes, so it should be easy to programmatically weed out duplicates.

**Reply by peppy** [reply]
importing from stable is required as it is an upgrade path

merging installs is not required and will not ever be implemented.

if you want to support this please make a separate stand alone project and maintain it yourself.
